.github/workflows/ci.yml

```diff
@@ -46,15 +47,42 @@ jobs:
         with:
           fetch-depth: 2
 
+      - name: cat settings profile
+        if: ${{ env.USE_STAGE == 'true' }}
+        run: |
+          cat $HOME/.m2/settings.xml
+
+      - name: Modify settings.xml
+        if: ${{ env.USE_STAGE == 'true' }}
+        run: |
+          sed -i 's/<\/settings>/<profiles><profile><id>local-repo<\/id><repositories><repository><id>central<\/id><url>https:\/\/repo.maven.apache.org\/maven2<\/url><releases><enabled>true<\/enabled><\/releases><snapshots><enabled>false<\/enabled><\/snapshots><\/repository><repository><id>staged-releases<\/id><url>https:\/\/repository.apache.org\/content\/groups\/staging\/<\/url><\/repository><\/repositories><pluginRepositories><pluginRepository><id>staged-releases<\/id><url>https:\/\/repository.apache.org\/content\/groups\/staging\/<\/url><\/pluginRepository><\/pluginRepositories><\/profile><\/profiles><activeProfiles><activeProfile>local-repo<\/activeProfile><\/activeProfiles><\/settings>/' $HOME/.m2/settings.xml
+
+      - name: cat settings profile
+        if: ${{ env.USE_STAGE == 'true' }}
+        run: |
+          cat $HOME/.m2/settings.xml
+
       - name: Compile
         run: |
           mvn clean compile -U -Dmaven.javadoc.skip=true -ntp
 
+      - name: Install JDK ${{ matrix.JAVA_VERSION }}
+        uses: actions/setup-java@v3
+        with:
+          java-version: ${{ matrix.JAVA_VERSION }}
+          distribution: 'zulu'
+
       - name: Install JDK 8
         uses: actions/setup-java@v3
         with:
           java-version: '8'
           distribution: 'zulu'
+
```

Comment:
why not specify a fixed copied settings.xml(rather than use the default file & rewrite it many times)